tests/test_verify.py

```diff
@@ -0,0 +1,33 @@
+import src.verify
+from src.verify import check
+from src.verify import main
+from pytest_mock import MockerFixture
+
+def test_verify():
+    N = (3, 4, 5, 1, 2, 3, 4, 9, 13, 0)
+    assert check(N, 4) == True
+    assert check(N, 10) == False
+
+def test_main_a(mocker: MockerFixture):
+    N = (3, 4, 5, 1, 2, 3, 4, 9, 13, 0)
+    mock_verify_return = True
+    mocker.patch(__name__ + '.check', return_value = mock_verify_return)
+    spy = mocker.spy(src.verify, "check")
+    
+    res = main(N, 4)
+
+    assert res == "The number 4 is present in the array"
+    assert spy.call_count == 1
+    assert spy.spy_return == mock_verify_return
```

Comment:
I didn't write this into the slides but you could  also use in addition
```python
assert spy.called_once_with(N, 4)
```

(you can also remove the call_count == 1 because the `called_once_with` already covers it)

p.s. = you could save the `4` into a separate `constant variable` to improve the code

Reply:
Ho provato la funzione che mi hai scritto ma sembra non esistere

Reply:
ricordavo male, si chiama `assert_called_once_with`, la puoi scrivere così
`spy.assert_called_once_with(N, 4)`

Reply:
p.s. vedi cosa ti segnala la pipeline